lib/lndclient/LndClient.ts

```diff
@@ -41,30 +41,32 @@ class LndClient extends BaseClient {
     const { disable, certpath, macaroonpath } = config;
 
     if (disable) {
-      this.setStatus(ClientStatus.DISABLED);
-    } else if (!fs.existsSync(certpath)) {
+      return;
+    }
+    if (!fs.existsSync(certpath)) {
       this.logger.error('could not find lnd certificate, is lnd installed?');
-      this.setStatus(ClientStatus.DISABLED);
-    } else if (!fs.existsSync(macaroonpath)) {
+      return;
+    }
+    if (!fs.existsSync(macaroonpath)) {
       this.logger.error('could not find lnd macaroon, is lnd installed?');
-      this.setStatus(ClientStatus.DISABLED);
-    } else {
-      this.uri = `${config.host}:${config.port}`;
-      const lndCert = fs.readFileSync(certpath);
-      this.credentials = grpc.credentials.createSsl(lndCert);
-
-      const adminMacaroon = fs.readFileSync(macaroonpath);
-      this.meta = new grpc.Metadata();
-      this.meta.add('macaroon', adminMacaroon.toString('hex'));
+      return;
```

Comment:
Not sure I agree here. The multiple return style makes the code much more readable. Constructors are not a special case in that sense. 

with current logic, the if-else flow has one thread so It is "manageable" but if you have to add another if to one of the "then" part it will become very complicated and uneasy to ready.

Unless this is critical for you I prefer to leave it like this.

Reply:
I am mainly just not used to seeing `return`s anywhere in constructors. I can't find anywhere that suggests it's a rule but it seems unconventional to me. To me they are equally readable, I'm not sure what exactly what you mean by adding another if, to me the difference is just between having `return` statements in all the `if` checks up front vs wrapping the rest of the constructor in an `else` bracket. It's not a big deal, but maybe someone else can chime in.

Reply:
@sangaman see this discussion on how if-else-if can become messy https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement
mainly the second answer.

Anyway, I will change it to single return and keep current structure (readable). Actually having an enhancment of checking validity of disable certpath and macaroonpath all at once.

Reply:
done